packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -593,10 +607,24 @@ const DatePickerInput = (props) => {
     }
     // increment current valid date by 1 day, if not valid date set date to tomorrow instead
     if (event.key === '=' || event.key === '+') {
-      if (validDate) {
-        inputDate = DateUtil.incrementDateByDay(inputDate, DateUtil.ISO_EXTENDED_DATE_FORMAT);
-      } else {
-        inputDate = DateUtil.incrementDateByDay(DateUtil.getCurrentDate(), DateUtil.ISO_EXTENDED_DATE_FORMAT);
+      if (inputType === 0) {
+        if (validDate) {
+          inputDate = DateUtil.incrementDateByDay(inputDate, DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        } else {
+          inputDate = DateUtil.incrementDateByDay(DateUtil.getCurrentDate(), DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        }
+      } else if (inputType === 1) {
+        if (validDate) {
+          inputDate = DateUtil.incrementMonthByMonth(inputDate, DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        } else {
+          inputDate = DateUtil.incrementMonthByMonth(DateUtil.getCurrentDate(), DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        }
+      } else if (inputType === 2) {
+        if (validDate) {
+          inputDate = DateUtil.incrementYearByYear(inputDate, DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        } else {
+          inputDate = DateUtil.incrementYearByYear(DateUtil.getCurrentDate(), DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        }
```

Comment:
There are Individual keydown event callbacks available for the 3 fields with code for the UP/down arrow key you can just add condition for +/- there and modify the code slightly instead of adding this. You can remove the older +/- logic from here and just add the +/- conditions to these functions. Individual field keydown callbacks below:
https://github.com/cerner/terra-framework/blob/91b1c58c141b3bcd6fae3875f3442807e2322a33/packages/terra-date-picker/src/DateInput.jsx#L455

https://github.com/cerner/terra-framework/blob/91b1c58c141b3bcd6fae3875f3442807e2322a33/packages/terra-date-picker/src/DateInput.jsx#L489

https://github.com/cerner/terra-framework/blob/91b1c58c141b3bcd6fae3875f3442807e2322a33/packages/terra-date-picker/src/DateInput.jsx#L525

Reply:
Existing functions to increment/ decrement the 3 fields won't set current date and it will increment/decrement only on focused field and remaining fields would be empty only. So i have simplified existing +/- functionality based on focused field - https://github.com/cerner/terra-framework/pull/1999/commits/860c744021f979dc3fef08c89fc1551a75905869